internal/controllers/machine/machine_controller_phases_test.go

```diff
@@ -1003,82 +1041,211 @@ func TestReconcileInfrastructure(t *testing.T) {
 			machine: &clusterv1.Machine{
 				ObjectMeta: metav1.ObjectMeta{
 					Name:      "machine-test",
 					Namespace: metav1.NamespaceDefault,
 				},
 				Spec: clusterv1.MachineSpec{
-					Bootstrap: clusterv1.Bootstrap{
-						ConfigRef: &corev1.ObjectReference{
-							APIVersion: "bootstrap.cluster.x-k8s.io/v1beta1",
-							Kind:       "GenericBootstrapConfig",
-							Name:       "bootstrap-config1",
-						},
-					},
 					InfrastructureRef: corev1.ObjectReference{
 						APIVersion: "infrastructure.cluster.x-k8s.io/v1beta1",
 						Kind:       "GenericInfrastructureMachine",
 						Name:       "infra-config1",
 					},
+					ProviderID:    ptr.To("test://something"),
+					FailureDomain: ptr.To("something"),
 				},
 				Status: clusterv1.MachineStatus{
-					BootstrapReady:      true,
 					InfrastructureReady: true,
-					NodeRef:             &corev1.ObjectReference{Kind: "Node", Name: "machine-test-node"},
+					Addresses: []clusterv1.MachineAddress{
+						{
+							Type:    clusterv1.MachineExternalIP,
+							Address: "1.2.3.4",
+						},
+					},
 				},
 			},
-			bootstrapConfig: map[string]interface{}{
-				"kind":       "GenericBootstrapConfig",
-				"apiVersion": "bootstrap.cluster.x-k8s.io/v1beta1",
+			infraMachine: map[string]interface{}{
+				"kind":       "GenericInfrastructureMachine",
+				"apiVersion": "infrastructure.cluster.x-k8s.io/v1beta1",
 				"metadata": map[string]interface{}{
-					"name":      "bootstrap-config1",
+					"name":      "infra-config1",
 					"namespace": metav1.NamespaceDefault,
 				},
-				"spec": map[string]interface{}{},
+				"spec": map[string]interface{}{
+					"providerID":    "test://id-1",
+					"failureDomain": "foo",
+				},
 				"status": map[string]interface{}{
-					"ready":          true,
-					"dataSecretName": "secret-data",
+					"ready": false,
+					"addresses": []interface{}{
+						map[string]interface{}{
+							"type":    "InternalIP",
+							"address": "10.0.0.1",
+						},
+						map[string]interface{}{
+							"type":    "InternalIP",
+							"address": "10.0.0.2",
+						},
+					},
 				},
 			},
-			infraConfig: map[string]interface{}{
-				"kind":       "GenericInfrastructureMachine",
-				"apiVersion": "infrastructure.cluster.x-k8s.io/v1beta1",
-				"metadata":   map[string]interface{}{},
-			},
-			expectResult: ctrl.Result{},
-			expectError:  true,
+			InfraMachineGetError: nil,
+			expectResult:         ctrl.Result{},
+			expectError:          false,
 			expected: func(g *WithT, m *clusterv1.Machine) {
 				g.Expect(m.Status.InfrastructureReady).To(BeTrue())
-				g.Expect(m.Status.FailureMessage).NotTo(BeNil())
-				g.Expect(m.Status.FailureReason).NotTo(BeNil())
-				g.Expect(m.Status.GetTypedPhase()).To(Equal(clusterv1.MachinePhaseFailed))
+				g.Expect(ptr.Deref(m.Spec.ProviderID, "")).To(Equal("test://id-1"))
+				g.Expect(ptr.Deref(m.Spec.FailureDomain, "")).To(Equal("foo"))
+				g.Expect(m.Status.Addresses).To(HaveLen(2))
```

Comment:
Can we also use this in other cases instead of `g.Expect(m.Status.Addresses).ToNot(BeNil())` please?